tests/sub_port_interfaces/conftest.py

```diff
@@ -55,14 +56,9 @@ def pytest_addoption(parser):
 
 @pytest.fixture(scope='module', autouse=True)
 def skip_unsupported_asic_type(duthost):
-    SUBPORT_UNSUPPORTED_ASIC_LIST = ["th2"]
-    vendor = duthost.facts["asic_type"]
-    hostvars = get_host_visible_vars(duthost.host.options['inventory'], duthost.hostname)
-    for asic in SUBPORT_UNSUPPORTED_ASIC_LIST:
-        vendorAsic = "{0}_{1}_hwskus".format(vendor, asic)
-        if vendorAsic in hostvars.keys() and duthost.facts['hwsku'] in hostvars[vendorAsic]:
-            pytest.skip(
-                "Skipping test since subport is not supported on {0} {1} platforms".format(vendor, asic))
+    SUBPORT_SUPPORTED_ASIC_LIST = ["td2"]
+    asic_type = duthost.get_asic_name()
+    py_require(asic_type in SUBPORT_SUPPORTED_ASIC_LIST, "Subport tests is not supported on {}".format(asic_type))
```

Comment:
What about other ASIC types which support those test cases? That means that test cases will be skipped on DUTs with those ASIC types, doesn't that?

Reply:
Thanks for the clarification, I will update it to skip for certain asics.